src/DataSourceProvider.vue

```diff
@@ -184,19 +187,24 @@ export default {
     loadDataSources(appId) {
       getDataSources(appId)
         .then(dataSources => {
-          const selectedDataSourceInDataSources = dataSources.some(dataSource => {
-            return dataSource.id === this.selectedDataSource.id;
-          });
+          if (this.widgetData.dataSourceId) {
+            const selectedDataSourceInDataSources = dataSources.some(dataSource => {
```

Comment:
Why aren't we using `.find()`?

```suggestion
            const selectedDataSourceInDataSources = dataSources.find(dataSource => {
```

Reply:
@tonytlwu Because we need only to determine if the selected data source in founded data sources.
So we use here true/false and we don't need an actual data source which we would found threw `.find`

Reply:
Rename it then.

```suggestion
            const selectedDataSourceFound = dataSources.some(dataSource => {
```